script/DeploymentConfig.s.sol

```diff
@@ -0,0 +1,51 @@
+//// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity >=0.8.19;
+
+import { Script } from "forge-std/Script.sol";
+
+contract DeploymentConfig is Script {
+    error DeploymentConfig_InvalidDeployerAddress();
+    error DeploymentConfig_NoConfigForChain(uint256);
+
+    struct NetworkConfig {
+        address deployer;
+    }
+
+    NetworkConfig public activeNetworkConfig;
+
+    address private deployer;
+
+    constructor(address _broadcaster) {
+        if (_broadcaster == address(0)) revert DeploymentConfig_InvalidDeployerAddress();
+        deployer = _broadcaster;
+        if (block.chainid == 31_337) {
+            activeNetworkConfig = getOrCreateAnvilEthConfig();
+        } else if (block.chainid == 11_155_111) {
+            activeNetworkConfig = getOrCreateSepoliaEthConfig();
+        } else if (block.chainid == 1442) {
+            activeNetworkConfig = getOrCreatePolygonZkevmConfig();
+        } else {
+            revert DeploymentConfig_NoConfigForChain(block.chainid);
+        }
+    }
+
+    function getOrCreateAnvilEthConfig() public view returns (NetworkConfig memory) {
+        return NetworkConfig({ deployer: deployer });
+    }
+
+    function getOrCreateSepoliaEthConfig() public view returns (NetworkConfig memory) {
+        return NetworkConfig({ deployer: deployer });
+    }
+
+    function getOrCreatePolygonZkevmConfig() public view returns (NetworkConfig memory) {
+        return NetworkConfig({ deployer: deployer });
+    }
```

Comment:
Unless there's a plan to actually provide different configurations pre network, we can probably remove these and simplify the constructor. But I'm also fine if we keep it for explicitness

Reply:
Leaving it in for explicitness if we want to change it!